packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -3,6 +3,50 @@ import { Stack } from './stack';
 import { Token } from './token';
 import { filterUndefined } from './util';
 
+/**
+ * An enum representing the various ARN formats that different services use.
+ */
+export enum ArnFormat {
+  /**
+   * This represents a format where there is no 'resourceName' part.
+   * This format is used for S3 resources,
+   * like 'arn:aws:s3:::bucket'.
+   * Everything after the last colon is considered the 'resource',
+   * even if it contains slashes,
+   * like in 'arn:aws:s3:::bucket/object.zip'.
+   */
+  NO_RESOURCE_NAME = 'arn:aws:service:region:account:resource',
+
+  /**
+   * This represents a format where the 'resource' and 'resourceName'
+   * parts are separated with a colon.
+   * Like in: 'arn:aws:service:region:account:resource:resourceName'.
+   * Everything after the last colon is considered the 'resourceName',
+   * even if it contains slashes,
+   * like in 'arn:aws:apigateway:region:account:resource:/test/mydemoresource/*'.
+   */
+  COLON_RESOURCE_NAME = 'arn:aws:service:region:account:resource:resourceName',
+
+  /**
+   * This represents a format where the 'resource' and 'resourceName'
+   * parts are separated with a slash.
+   * Like in: 'arn:aws:service:region:account:resource/resourceName'.
+   * Everything after the separating slash is considered the 'resourceName',
+   * even if it contains colons,
+   * like in 'arn:aws:cognito-sync:region:account:identitypool/us-east-1:1a1a1a1a-ffff-1111-9999-12345678:bla'.
+   */
+  SLASH_RESOURCE_NAME = 'arn:aws:service:region:account:resource/resourceName',
+
+  /**
+   * This represents a format where the 'resource' and 'resourceName'
+   * parts are seperated with a slash,
+   * but there is also an additional slash after the colon separating 'account' from 'resource'.
+   * Like in: 'arn:aws:service:region:account:/resource/resourceName'.
+   * Note that the leading slash is _not_ included in the parsed 'resource' part.
+   */
+  SLASH_RESOURCE_NAME_AND_RESOURCE = 'arn:aws:service:region:account:/resource/resourceName',
```

Comment:
```suggestion
  SLASH_RESOURCE_AND_NAME = 'arn:aws:service:region:account:/resource/resourceName',
```

By our own definition

Reply:
Done.